### PR TITLE
Footer i18n: Update footer branding to be well supported in multiple languages

### PR DESCRIPTION
**Pull request by WBerredo**
Related to #77916

## Proposed Changes

Replace `Automattic` text for a variable containing Automattic SVG  in order to avoid the company name being translated

CleanShot 2023-06-12 at 12 33 55@2x

## Testing Instructions

**This branch**
* On a logged-out tab
* Go to a page that displays the footer. Ex: /theme/miniature
* Change the language (Ex: `Español`) and confirm the name Automattic is not being translated

**Prod**
* On a logged-out tab
* Go to a page that displays the footer. Ex: /theme/miniature
* Change the language (Ex: `Español`) and confirm the name Automattic  is being translated to `automático`

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/wpcom-template-parts/src/utils.ts [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
+		translate( 'An %(Automattic)s contraption', options ),
+		translate( 'An %(Automattic)s creation', options ),
+		translate( 'An %(Automattic)s experiment', options ),
+		translate( 'An %(Automattic)s invention', options ),
+		translate( 'An %(Automattic)s joint', options ),
+		translate( 'An %(Automattic)s medley', options ),
+		translate( 'An %(Automattic)s opus', options ),
+		translate( 'An %(Automattic)s production', options ),
+		translate( 'An %(Automattic)s ruckus', options ),
+		translate( 'An %(Automattic)s thingamajig', options ),
```
Comment (jsnajdr) [inline packages/wpcom-template-parts/src/utils.ts]: This is a good idea, to turn "Automattic" into a variable. Having this, you could pass the SVG directly into the `translate` function:
```js
function AutomatticBrand() {
  return <>
    <span hidden>Automattic</span>
    <svg><path d="draw the letters"/></svg>
  </>;
}

function getAutomatticBranding( translate, brand ) {
  return translate( 'An %(Automattic) thing', { args: { Automattic: brand } } );
}

getAutomatticBranding( translate, <AutomatticBrand /> );
```
With interpolated translation string, you no longer need to split manually into prefix and suffix.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8035323

Thank you @WBerredo for including a screenshot in the description! This is really helpful for our translators.

File: packages/wpcom-template-parts/src/utils.ts [hunk 2]
```diff
@@ -1,24 +1,30 @@
...
 	];
 
-	const branding = automatticRoger[ Math.floor( Math.random() * ( 10 - 0 + 1 ) + 0 ) ];
+	const branding = automatticRoger[ Math.floor( Math.random() * ( 10 - 0 + 1 ) + 0 ) ]?.toString();
```
Comment (jsnajdr) [inline packages/wpcom-template-parts/src/utils.ts]: I'm wondering what these `±0` arithmetics are supposed to achieve. To select a random array element, seems to me that simple `roger[ Math.floor( Math.random() * roger.length ) ]` is sufficient.

Also, once you avoid the `.split` call and use interpolation to insert the A8c logo, the `.toString` call is no longer needed. In theory `branding` can be a HTML/JSX markup then.

Comment (WBerredo) [reply packages/wpcom-template-parts/src/utils.ts]: > I'm wondering what these ±0 arithmetics are supposed to achieve. To select a random array element, seems to me that simple roger[ Math.floor( Math.random() * roger.length ) ] is sufficient.

Agreed, I'm also not sure why are they needed, but I suppose it must be needed as it is a non-trivial addition.

File: packages/wpcom-template-parts/src/utils.ts (hunk 1 above)
Comment (WBerredo) [reply packages/wpcom-template-parts/src/utils.ts:21]: I really liked this solution and tried to implement it, but apparently, when interpolating the translation with a markup it always generates a different signature.

CleanShot 2023-06-12 at 12 59 15@2x

Comment (WBerredo) [reply packages/wpcom-template-parts/src/utils.ts]: This is a great idea, ty.

I've implemented it at https://github.com/Automattic/wp-calypso/pull/78018/commits/c9c94345dac6275b4d39bf5f1991fe01018c6a41

File: packages/wpcom-template-parts/src/universal-footer-navigation/index.tsx [hunk 3]
```diff
@@ -477,7 +464,9 @@ const UniversalNavbarFooter = ( {
 	const isEnglishLocale = useIsEnglishLocale();
 	const pathNameWithoutLocale =
 		currentRoute && removeLocaleFromPathLocaleInFront( currentRoute ).slice( 1 );
-	const [ automatticBranding, setAutomatticBranding ] = useState( { article: '', noun: '' } );
+	const [ automatticBranding, setAutomatticBranding ] = useState< React.ReactChild >(
```
Comment (jsnajdr) [inline packages/wpcom-template-parts/src/universal-footer-navigation/index.tsx]: The `React.ReactChild` type (or rather `React.JSX.Element`) will be automatically inferred from the default value, doesn't need to be specified.

File: packages/wpcom-template-parts/src/utils.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import { translate as translateMethod } from 'i18n-calypso';
+
+export const getAutomatticBrandingNoun = ( translate: typeof translateMethod ) => {
+	const options = {
+		description:
+			'Branding to be shown on the Footer of the page, Automattic\'s variable will always contains the word "Automattic"',
```
Comment (jsnajdr) [inline packages/wpcom-template-parts/src/utils.tsx]: This `options.description` field, will it be correctly extracted by the `i18n` tool? I have some doubts about that. We might have to repeat it verbatim for every variant, however ugly that is.

**Review by epeicher — APPROVED** [review]
I have tested it, and it works as described. Using the Calypso link and changing the language to _Español_, I cannot see the word `Automattic` being translated while it is translated in wordpress.com. 

LGTM! 🚢 
CleanShot 2023-06-13 at 13 35 53@2x

**Review by jsnajdr — APPROVED** [review]
Looks good now :+1:

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8035323).

**Pull request merged**